noir/noir-repo/acvm-repo/brillig_vm/src/lib.rs

```diff
@@ -290,7 +290,7 @@ impl<'a, B: BlackBoxFunctionSolver> VM<'a, B> {
                 let source = self.memory.read_ref(*source_pointer);
                 // Use our usize source index to lookup the value in memory
                 let value = &self.memory.read(source);
-                self.memory.write(*destination_address, *value);
+                self.memory.write(*destination_address, value.clone());
```

Comment:
```suggestion
                self.memory.write(*destination_address, value);
```
And remove the `&` in `let value = &self.memory.read(source)` the line above.